### DISCUSSION TITLE
Strategies for .env generation on CD process.

### DISCUSSION THREAD

**Discussion by karlkras** [brefphp/bref#1168] [Support] [answered]
We're setting up our automated bref deployment from using GitHub actions.
Something that doesn't seem to get much mention is how to dynamically build/populate the .env to be deployed with sensitive information to the target environments.
I'm assuming we put this information in GitHub  key storage and add a build step to do something with accessing and inserting the data accordingly, but since I see no mention of this being a concern, I have to wonder if I'm missing something and making this more difficult than need be?
Are there any examples or specific instructions on best practices for handling this?

**Comment by t-richard** [comment, marked as answer]
When in the cloud, you would rpobably want to avoid using the `.env` file.

Instead, use environment variables as described here https://bref.sh/docs/environment/variables.html

Those can be used in combination with github action like this

```
provider:
    # ...
    environment:
        MY_VARIABLE: ${env:MY_VARIABLE} 
```

What is used in `${env:MY_VARIABLE}` can be the name of the variable defined in github actions secrets

**Reply by karlkras** [reply]
Well I'll be darned. I thought that was provided as a possible choice rather than preferred. But it makes perfect sense and straight forward. Thanks!
